packages/component-library/src/components/table-and-list/mt-data-table/mt-data-table.vue

```diff
@@ -1446,6 +1592,107 @@ export default defineComponent({
       }
     };
 
+    function removeFilter(id: string) {
+      const updatedFilters = props.appliedFilters.filter((filter) => filter.id !== id);
+
+      emit("update:appliedFilters", updatedFilters);
```

Comment:
Personal thing: I would prefer events which are more explicit, e.g. in this case just a event which tells that a filter was removed and the id.

The reason for this is that with sending all filers directly it is harder from the outside to get the changes. Because in most cases you won't to replace the filter data in the parent component with the newly emitted one because it is very likely that some reactivity deps wouldn't get triggered that way. So the parent component need to create a diff between the new filters from the event and update them on the real data then.

So it would be easier to maintain and less error prone if they would be more explicit. The dumber the base component, the easier to maintain.